Tycherious Script Update

mgord54 opened the pull request:
30/6/2025 Update Change list:

Fixed the civil war debuff of Abbasid and Epirus does not get removed when the civil war ends. 
Fixed Seljuks faction attribute bug (+2 experience for missile cav recruit now working)
Fixed muhtasib minister bug ( +2 sanitation and +%10 commerce traits now working)
Fixed Startup menu bug saying ‘missing pack files’, now this feature is working again and will notify if a pack is missing properly. 
HRE Voting System with same mechanics - but reworked functions; see:(transparent HRE Mechanics)
HRE Election System – Transitions Occur for title and Frankfurt (Empire Capitol).
HRE dynamic naming – ‘Holy Roman Empire’ faction name transition fixed. 
Fixed the three unique HRE dilemmas that occur for the Emperor faction that shape loyal/disloyal relationships within the HRE system.
Fixed pretender system, the Pope can again elect pretender catholic factions that are able to engage with the HRE mechanics. 
Added Election Event Message Pop-up with image and text description after a new Emperor is elected, to inform the player. 
Liberate member state as Holy Roman Emperor successfully organises automatic military alliance and loyal HRE member state condition.
HRE minor Faction, Puppet State Fixed.
HRE Reform #1 now correctly limited to 6 Prince Elector States + Emperor as voters in HRE Panel instead of having 9.

mgord54 committed:
Update

fedacking commented on campaigns/main_attila/mechanics/hre/mechanics_hre_events.lua:
```diff
@@ -1,7 +1,7 @@
 -----------------------------------------------------------------------------------------------------------------------------------------------
 --
--- 	MEDIEVAL KINGDOMS 1212 AD - MECHANICS: HOLY ROMAN EMPIRE EVENTS
+-- 	MEDIEVAL KINGDOMS 1212 - MECHANICS: HOLY ROMAN EMPIRE EVENTS
```
Rip AD

fedacking commented on campaigns/main_attila/mechanics/hre/mechanics_hre_events.lua:
```diff
@@ -11,50 +11,36 @@
 local hre_events_timer = -1;
 
 function mkHRE:Add_Event_Listeners(emperor_key)
-    DebugLog("Add_Event_Listeners: Starting listener setup.")
```
We should fix it so that DebugLog is no longer your local folder hardcoded

mgord54 replied:
To be honest that's just a noob mistake of mine I think, I have tried to do that but without a hardcoded folder I couldn't get debugging to work otherwise. 

I would really like to remove the debugging at some point but I don't want to take it out and then find bugs that I could of used it to fix easier, ya know.

fedacking commented on campaigns/main_attila/mechanics/hre/mechanics_hre_factions.lua:
```diff
@@ -11,8 +11,8 @@
-local hre_imperial_authority_gain_per_region = 0.2; -- Imperial Authority gain per region in the HRE.
-local hre_faction_state_change_cooldown = 10; -- How many turns before a faction's state can change after it has been changed?
+local hre_imperial_authority_gain_per_region = 0.3; -- Imperial Authority gain per region in the HRE.
+local hre_faction_state_change_cooldown = 2; -- How many turns before a faction's state can change after it has been changed?
```
I would make it slightly higher, in case you force it with something you want it to last longer. Maybe have different types of cooldown?

mgord54 replied:
Oh, I really love this idea. Did you wanna make the change and I'll test it in game?

fedacking commented on campaigns/main_attila/mechanics/hre/mechanics_hre_factions.lua:
```diff
@@ -445,74 +435,70 @@ function TimeTrigger_HRE_Factions(context)
-		end
+    -- Handle vassalized "puppet" state
+    if faction_state ~= "puppet" then
+        if emperor_faction:is_human()
```
Only human puppets?

mgord54 replied:
Good find, there's no reason for the 'is_human' clause that I can remember. I'll update this and test it out.

fedacking commented on campaigns/main_attila/mechanics/hre/mechanics_hre_factions.lua:
```diff
@@ -445,74 +435,70 @@ function TimeTrigger_HRE_Factions(context)
-			local chance = cm:random_number(5);
+        -- Dynamic stance-based loyalty adjustments
+        if turn_number > 1 then
+            if stance >= 3 then
```
Stance >= 3 is really hard to get. It should be 2 imo.

mgord54 replied:
You should add this change yourself as well and just send the pack file over. 
I know what you mean and agree but it's real important we don't make it too easy for a faction to rule.

fedacking commented on campaigns/main_attila/mechanics/hre/mechanics_hre_factions.lua:
```diff
@@ -445,74 +435,70 @@ function TimeTrigger_HRE_Factions(context)
+        if turn_number > 1 then
+            if stance >= 3 then
+                faction_state = "loyal"
+            elseif stance > 0 then
```
this should be stance >= 0

mgord54 replied:
Got it! Thank you.

fedacking commented on campaigns/main_attila/mechanics/hre/mechanics_hre_factions.lua:
```diff
@@ -445,74 +435,70 @@ function TimeTrigger_HRE_Factions(context)
-			self:Set_Faction_State(faction_name, "neutral", false);
-		end
-	end
+            -- Optional: small drift from neutral into 'ambitious'
```
Maybe we should make it so that ambitious states tend to remain ambitious, but it will be a harder logic. Maybe change it when their ruler dies? spitballing here

mgord54 replied:
I love the idea, the implementation sounds like a headache for me, if you're up for it give it a whirl in your version and I'll send it to our little hoard of testers. ;)

fedacking reviewed (changes requested):
Left some feedback

mgord54 committed:
Update mechanics_hre_events.lua

Updated with Fedackings advice.

mgord54 committed:
Merge pull request #3 from mgord54/mgord54-patch-1

Update mechanics_hre_events.lua

mgord54 committed:
Update mechanics_hre_reforms.lua

Factions updated:
Fixed only human emperor can make puppets, now AI should be able to.

mgord54 committed:
Update mechanics_hre_regions.lua

Reforms Update

mgord54 committed:
Update mechanics_hre_factions.lua

mgord54 committed:
Update mechanics_hre_regions.lua

mgord54 committed:
Update mechanics_hre_reforms.lua

mgord54 commented:
> Left some feedback

I resolved some things you mentioned. 

I hope I tackled that right... GitHub is a little overwhelming but the more I learn about it the better it gets.
